subgraph/src/utils.ts

```diff
@@ -0,0 +1,384 @@
+import {
```

Comment:
can we please organise this "utils" file into dedicated files with descriptive names and tests?

Reply:
Reorganized into dedicated folders and files. Tested respective functions on package repository.